src/main/java/com/upserve/uppend/lookup/LookupData.java

```diff
@@ -419,12 +422,26 @@ private int getMetaDataGeneration() {
         return metaDataGeneration.get();
     }
 
+    private static LongAdder longAdderWithInitialValue(long initialValue) {
+        LongAdder la = new LongAdder();
+        la.add(initialValue);
+        return la;
+    }
+
     private LongAdder getMetaHitCount() {
-        return Optional.ofNullable(atomicMetadataRef.get()).map(md -> md.hitCount).orElse(new LongAdder());
+        return getMetaHitCount(0l);
+    }
+
+    private LongAdder getMetaHitCount(long initialValue) {
+        return Optional.ofNullable(atomicMetadataRef.get()).map(md -> md.hitCount).orElse(longAdderWithInitialValue(initialValue));
```

Comment:
These are getting to complicated.
I think we want to pass the actual longadder around. Kust make one of them and keep passing it to the new Metadata instance to avoid any race conditions.
I think we can leave these methods available as they were, only for getting the atomicMetadataRef or empty case.

Reply:
```java

Reply:
Something like this?
```java
    private static LongAdder theLongAdder = new LongAdder();
    
    private LongAdder getMetaHitCount() {
        return Optional.ofNullable(atomicMetadataRef.get()).map(md -> md.hitCount).orElse(theLongAdder));
    }

    private LongAdder getMetaMissCount(long initialValue) {
        return Optional.ofNullable(atomicMetadataRef.get()).map(md -> md.missCount).orElse(theLongAdder);
    }
```

Reply:
no, wait, it needs to be set to the value of either `prevHitCount` or `prevMissCount`

Reply:
Hmmm - static is going to be one for the every instance of the class right (Polyglot brain)?

We definitely want one counter instance for each instance of LookupData/LookupMetadata

Reply:
Resolved.